functions/index.js

```diff
@@ -34,28 +40,61 @@ exports.createNewTask = functions.https.onRequest((request, response) => {
         console.log(status);
         console.log(category);
         console.log(createNewTaskSprintNumber);
+        console.log(storyPointNumber);
 
-        db.collection(fullSprintId).doc(category).set({
-                Title: title,
-                Description: des,
-                Priority: priority,
-                Difficulty: difficulty,
-                Creator: creator,
-                Assignee: assignee,
-                ET: estimatedTime,
-                Status: status,
-                Category: category,
-                LogWorkTotalTime: loggedWorkTotalTime,
-                WorkDone: workDone,
-                CreateNewTaskSprintNumber: createNewTaskSprintNumber
+        db.collection("Main").doc("RawData").get().then((doc) => {
+
+                totalDevelopmentTask = doc.data().TotalDevelopmentTask + 1;
+                totalBusinessTask = doc.data().TotalBusinessTask + 1;
+                totalMarketingTask = doc.data().TotalBusinessTask + 1;
+                totalNumberOfTask = (totalDevelopmentTask + totalBusinessTask + totalMarketingTask);
+
+                if (category === "Development") {
+                    taskId = category[0] + totalDevelopmentTask;
+                } else if (category === "Business") {
+                    taskId = category[0] + totalBusinessTask;
+                } else {
+                    taskId = category[0] + totalMarketingTask;
+                }
+
+                console.log(taskId);
+
+                var setDataPromise = db.collection(fullSprintId).doc(taskId).set({
+                    Title: title,
+                    Description: des,
+                    Priority: priority,
+                    Difficulty: difficulty,
+                    Creator: creator,
+                    Assignee: assignee,
+                    ET: estimatedTime,
+                    Status: status,
+                    Category: category,
+                    LogWorkTotalTime: loggedWorkTotalTime,
+                    WorkDone: workDone,
+                    CreateNewTaskSprintNumber: createNewTaskSprintNumber,
+                    StoryPointNumber: storyPointNumber
+                });
+                return Promise.resolve(setDataPromise);
             })
-            .then(() => {
-                var work = { data: "working" }
+            .then(function(setDataPromise) {
+                var updateSetDataPromise = db.collection("Main").doc("RawData").update({
+                    TotalDevelopmentTask: totalDevelopmentTask,
+                    TotalBusinessTask: totalBusinessTask,
+                    TotalMarketingTask: totalMarketingTask,
+                    TotalNumberOfTask: totalNumberOfTask
+
+                });
+                return Promise.resolve(updateSetDataPromise);
+            })
+            .then((updateSetDataPromise) => {
+                var result = { data: "working" }
```

Comment:
working can be replaced with OK,

Please make the result variable out of the .then scope

Please look the comment into line no : 94 and 95. for example.

Reply:
yeah okay